src/lib/telemetry.js

```diff
@@ -0,0 +1,39 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+import Glean from '@mozilla/glean/web';
+
+const APP_NAME = "debug-ping-view";
+const APP_TELEMETRY_PREFERENCE_STORAGE_KEY = "telemetryEnabled";
+
+function isTelemetryEnabled() {
+  return navigator.doNotTrack !== "1";
+}
+
+function setAppTelemetryPreferenceInStorage(telemetryPreferenceValue) {
+    sessionStorage.setItem(APP_TELEMETRY_PREFERENCE_STORAGE_KEY, telemetryPreferenceValue)
+}
+
+function getAppTelemetryPreferenceFromStorage() {
+  return sessionStorage.getItem(APP_TELEMETRY_PREFERENCE_STORAGE_KEY);
+}
+
+// Initialize Glean.
+export function initTelemetry() {
+  const telemetryEnabled = isTelemetryEnabled();
+  setAppTelemetryPreferenceInStorage(telemetryEnabled.toString());
+  Glean.initialize(APP_NAME, telemetryEnabled, { maxEvents: 1 });
+}
+
+// Update Glean upload settings if user's consent to telemetry changes
+export function updateTelemetry() {
+  const appTelemetryPreference = getAppTelemetryPreferenceFromStorage();
+  const telemetryEnabled = isTelemetryEnabled();
+  if (appTelemetryPreference !== telemetryEnabled.toString()) {
```

Comment:
Why do you do this? Glean should handle this internally.

Reply:
@Dexterp37 Probably, I didn't understand your comment. Do you mean Glean will detect the change in `navigator.doNotTrack` property value internally and disable/enable upload automatically?

Reply:
No, I mean you are checking whether or not the current state of `navigator.doNotTrack` is different than a state you are storing in `appTelemetryPreference`. I believe this is done by Glean.js itself (@rosahbruno ?). If not, it should still be done by Glean.js (as the [other sdks do that](https://searchfox.org/glean/rev/4266afeed9e673aba64cb35ad0f6f5f8db11ac35/glean-core/src/lib.rs#783)).

Reply:
So this was my idea as well in the begining that we can directly call `Glean.setUploadEnabled(isTelemetryEnabled())` without doing this check. At most it will be a no-op if `navigator.doNotTrack` didn't change compared to the last time.

But Bruno was of the opinion that we can just avoid calling the `Glean.setUploadEnabled` API every single time we change pages. Even though it ends up being a no-op, it still does two additional checks first no matter what. If we store it in storage, we can easily just pull out that value from the browser API and then check if we see any difference. Also from a readability standpoint, calling `setUploadEnabled` on every single route navigation is a bit confusing.

Reply:
Documenting for completeness.

Another question raised by @Dexterp37 is _why do you need to call setUploadEnabled() given that we already initialize glean on every single route?_

What I found out yesterday was that having `Glean.initialize` call in [index.js](https://github.com/mozilla/debug-ping-view/blob/main/src/index.js#L18) isn't resulting into reinitializing Glean on every single route.

So, my idea was to move the `Glean.initialize()` call from index.js to [SecuredRoute component](https://github.com/mozilla/debug-ping-view/blob/main/src/components/SecuredRoute.js) as part of the [current PR in review](https://github.com/mozilla/debug-ping-view/pull/85#discussion_r1327003674). But the result of the discussions with @rosahbruno were that that `index.js` file is still a better place for keeping `Glean.initialize` API call.

Reply:
Just documenting what @rosahbruno clarified in chat.

Subsequent calls to `Glean.initialize()` API even with different parameters will be a [no-op as per the code](https://github.com/mozilla/glean.js/blob/main/glean/src/core/glean/sync.ts#L160C6-L167). Hence, moving this call within `SecuredRoute` component will not work and we will still have to use `setUploadEnabled()` to enable/disable telemetry.